jenkins/Hono-Release-Pipeline.groovy

```diff
@@ -0,0 +1,142 @@
+    try {
+        checkOut()
+        setReleaseVersionAndBuild()
+        commitAndTag()
+        setNextVersion()
+        commitAndPush()
+        deploy()
+        utils.publishJavaDoc('target/site/apidocs', true)
+        currentBuild.result = 'SUCCESS'
+    } catch (err) {
+        currentBuild.result = 'FAILURE'
+        echo "Error: ${err}"
+    }
+    finally {
+        echo "Build status: ${currentBuild.result}"
+        utils.notifyBuildStatus()
+    }
+}
+
+/**
+ * Checks out the specified branch from git repo
+ *
+ */
+def checkOut() {
+    stage('Checkout') {
+        echo "Check out branch: origin/${params.BRANCH}"
+        checkout([$class                           : 'GitSCM',
+                  branches                         : [[name: "origin/${params.BRANCH}"]],
+                  doGenerateSubmoduleConfigurations: false,
+                  extensions                       : [[$class: 'WipeWorkspace'],
+                                                      [$class: 'LocalBranch']],
+                  userRemoteConfigs                : [[credentialsId: 'bebef3c5-da22-425c-8554-f62d0b3a9608', url: 'ssh://github.com/eclipse/hono.git']]])
+    }
+}
+
+/**
+ * Set version to RELEASE_VERSION and build using maven
+ *
+ */
+def setReleaseVersionAndBuild() {
+    stage('Build') {
+        withMaven(maven: 'apache-maven-latest', jdk: 'jdk1.8.0-latest', options: [jacocoPublisher(disabled: true), artifactsPublisher(disabled: true)]) {
+            sh "mvn versions:set -DgenerateBackupPoms=false -DnewVersion=${params.RELEASE_VERSION}"
+            sh 'mvn clean install javadoc:aggregate -Dmaven.test.failure.ignore=false -DenableEclipseJarSigner=true -DsnapshotDependencyAllowed=false -Ddocker.skip.build=true'
+        }
+    }
+}
+
+/**
+ * Commit and tag to RELEASE_VERSION
+ *
+ */
+def commitAndTag() {
+    stage("Commit and tag release version") {
+        sh ''' 
+            git add pom.xml \\*/pom.xml
+            git commit -m "Release ${RELEASE_VERSION}"
+            git tag ${RELEASE_VERSION}
+           '''
+    }
+
+}
+
+/**
+ * Set version to NEXT_VERSION using maven
+ *
+ */
+def setNextVersion() {
+    stage("Set next version") {
+        withMaven(maven: 'apache-maven-latest', jdk: 'jdk1.8.0-latest', options: [jacocoPublisher(disabled: true), artifactsPublisher(disabled: true)]) {
+            sh "mvn versions:set -DallowSnapshots=true -DgenerateBackupPoms=false -DnewVersion=${NEXT_VERSION}"
+        }
+    }
+}
+
+/**
+ * Commit and push to the Hono repo
+ *
+ */
+def commitAndPush() {
+    stage("Commit and push") {
+        sh ''' 
+            git add pom.xml \\*/pom.xml
+            git commit -m "Bump version to ${NEXT_VERSION}"
+            git push --all ssh://git@github.com/eclipse/hono.git && git push --tags ssh://git@github.com/eclipse/hono.git
+           '''
+    }
+}
+
+/**
+ * Copy the artifacts so that the binaries are available for download
+ *
+ */
+def deploy() {
+    stage("Deploy Artifacts") {
```

Comment:
IMHO this should be rather named *Copy Artifacts` because *deploy* has a different meaning in the context of Maven (where it is used to upload the artifacts to a repository).